app/flatpak-builtins-build-export.c

```diff
@@ -1013,6 +1015,16 @@ flatpak_builtin_build_export (int argc, char **argv, GCancellable *cancellable,
     g_variant_dict_insert_value (&metadata_dict, OSTREE_COMMIT_META_KEY_ENDOFLIFE,
                                  g_variant_new_string (opt_endoflife));
 
+  if (opt_endoflife_rebase && *opt_endoflife_rebase)
+    {
+      g_auto(GStrv) eol_rebase_ref_parts = flatpak_decompose_ref (opt_endoflife_rebase, error);
```

Comment:
Everything else says the rebase is meant to be an ID, but here you parse it as a ref. Which is it?

Reply:
Hmm, it seems to me like the *key* needs to be a ref, but it makes a lot of sense for the argument to be an id. We should probably accept an id, and reconstruct a full ref from the other parts of the normal ref.